adlfs/transfer.py

```diff
@@ -0,0 +1,368 @@
+# -*- coding: utf-8 -*-
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+High performance multi-threaded module to up/download
+
+Calls method in `core` with thread pool executor to ensure the network
+is used to its maximum throughput.
+
+Only implements upload and download of (massive) files and directory trees.
+"""
+from collections import namedtuple
+from concurrent.futures import ThreadPoolExecutor
+import logging
+import multiprocessing
+import os
+import pickle
+import threading
+import time
+import uuid
+
+from adlfs.utils import clamp
+
+MAX_THREADS = 64
```

Comment:
As long as it is obvious to a user/caller how to get a higher value (and easy in a CLI implementation) then I am ok with having an internal default, I just want to make sure it is well documented so users understand what the intention is. As it was a user would think that adding more threads wouldn't give value or would not work, potentially.

---
In reply to: [78975720](https://github.com/Azure/azure-data-lake-store-python/pull/64#discussion_r78975720) [](ancestors = 78975720)